### ISSUE TITLE
Android: stopped AudioRecorder is resurrected by onErrorAfterClose on audio route change (mic stays hot; SIGSEGV in oboe fireDataCallback on 0.12.x)

### ISSUE THREAD

**Issue by christian-apollo** [software-mansion-labs/react-native-audio-api#1088]
## Summary

`AndroidAudioRecorder` has a set of related lifecycle bugs around oboe's `ErrorDisconnected` handling that (a) turn the microphone back on after `stop()` whenever the audio route changes, and (b) on 0.12.x crash the app with a native SIGSEGV in `oboe::AudioStream::fireDataCallback`.

## Environment

- react-native-audio-api **0.12.2** (crash reproduced in production); the same logic bugs are still present on current `main`
- Android, arm64-v8a

## Production crash (Crashlytics tombstone, 0.12.2)

```
#00 pc 0x2e9d0  oboe::AudioStream::fireDataCallback(void*, int)  [liboboe.so]
#01 pc 0x2c6d8  oboe::AudioStreamAAudio::callOnAudioReady(AAudioStreamStruct*, void*, int)  [liboboe.so]
#02 pc 0x48b90  aaudio::AudioStream::maybeCallDataCallback(void*, int)  [libaaudio_internal.so]
#03 pc 0x55090  aaudio::AudioStreamLegacy::callDataCallbackFrames(unsigned char*, int)
#04 pc 0x545b4  aaudio::AudioStreamLegacy::onMoreData(android::AudioRecord::Buffer const&)
#05 pc 0x6d9a4  android::AudioRecord::processAudioBuffer()  [libaudioclient.so]
#06 pc 0x6d0bc  android::AudioRecord::AudioRecordThread::threadLoop()
```

## Root causes (all in `AndroidAudioRecorder.cpp`)

1. **`stop()` never closes the stream** — it only calls `mStream_->requestStop()`. The stream stays open with its data/error callbacks registered indefinitely.

2. **`onErrorAfterClose()` unconditionally restarts the stream** on `ErrorDisconnected`, with no lock and no state check. Any audio route change (Bluetooth headset connecting/disconnecting, speaker switch) on a *stopped* recorder reopens the stream and flips `state_` back to `Recording` — the mic silently turns back on after `stop()`. On `main`, where the stream holds a `shared_ptr` to the recorder (`shared_from_this()`), the resurrected stream also keeps the recorder alive forever after JS drops it (reference cycle: recorder → `mStream_` → recorder), so this is a privacy bug *and* a leak.

3. **Unsynchronized `mStream_` access across threads.** oboe's error callback runs on a *detached* thread (`AudioStreamAAudio::internalErrorCallback` → `std::thread(...).detach()`), and `onErrorAfterClose` → `cleanup()` resets/reopens `mStream_` there while `start()`/`stop()`/the destructor mutate it on the JS thread with no common lock. On 0.12.2, where the stream holds a *raw* pointer to the recorder (`setDataCallback(this)`), the interleaving

   - JS GC destroys the recorder host object → destructor reads `mStream_` just as the error thread's `cleanup()` reset it → skips the close → object freed
   - error thread then reopens a stream whose data callback points at the freed recorder

   produces exactly the `fireDataCallback` SIGSEGV above on the next mic buffer.

4. Because the error thread is detached it can also arrive **late**: if `stop()`/`start()` already replaced the stream, the handler closes the *healthy new* stream and reopens it without re-running `prepare()`.

5. `isRecording()` dereferences `mStream_` without a null check; the error thread can reset it concurrently (and `pause()` calls `mStream_->pause(0)` based on that check).

## Reproduction sketch

1. `recorder.onAudioReady(...)`; `recorder.start()`; `recorder.stop()`; drop the JS reference.
2. Connect or disconnect a Bluetooth audio device → mic recording restarts (verifiable via the mic-in-use indicator / `dumpsys audio`).
3. On 0.12.x, keep toggling the audio route while the GC collects the old recorder → eventual SIGSEGV in `oboe::AudioStream::fireDataCallback`.

## Suggested fix

- `stop()` fully closes and resets the stream (also breaks the `shared_from_this` reference cycle); `start()` already reopens on demand.
- `onErrorAfterClose()` takes the same locks as `start()`/`stop()`, bails when the recorder is idle, and bails when the erroring stream is no longer `mStream_`.
- Null-check `mStream_` in `isRecording()`; take the recorder locks in `pause()`/`resume()`.

PR with this fix incoming. We've been running it in production as a patch-package patch on 0.12.2 (compiled for arm64-v8a; lock ordering verified against oboe 1.9.3 — the data callback only ever `try_lock`s the recorder mutexes and error threads are detached, so closing under the locks cannot deadlock).
